File: 0053-maximum-subarray/solution.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if not nums:
            return 0
        max_sum = nums[0]
        current_sum = nums[0]
        for i in range(1, len(nums)):
            current_sum = max(current_sum + nums[i], nums[i])
            max_sum = max(max_sum, current_sum)
        return max_sum

File: 0053-maximum-subarray/solution.rs
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut best: i32 = nums[0];
        let mut sum: i32 = 0;
        for index in 0..nums.len() {
            sum = nums[index].max(sum + nums[index]);
            best = best.max(sum);
        }
        best
    }
}
